### DISCUSSION TITLE
Using n8n and Node Red for AI support in Trilium Next

### DISCUSSION THREAD

**Discussion by elaiel** [TriliumNext/Trilium#6036] [Ideas]
Hi everyone,

I'm not sure if this topic fits here, but I’m working on adding AI functionality to note creation in Trilium Next using automation flows and the Trilium Next API. Everything works smoothly with Node-RED; I can create, read, and modify notes. However, I’m running into issues with n8n (https://n8n.io/).

My goal is to use Telegram voice recordings as a basis, convert them to text, categorize them (e.g., notes, ideas, to-do's), and add them as new notes or append them to existing ones. Creating new notes via HTTP POST requests works fine, but retrieving and modifying notes through n8n has been challenging. Although there's a connection, I’m not receiving any data, nor am I able to append anything to existing notes. Since Node-RED handles this without issues, I'm stumped. n8n logs have no error messages or issues.

Has anyone successfully connected n8n to Trilium Next or managed similar workflows?

Thanks for any insights!

**Comment by meichthys** [comment]
I haven't seen anyone talk about using Trilium with n8n yet, but I hope someone chimes in because it sounds like an interesting use case!

**Reply by meichthys** [reply]
There was this issue raised on the original Trilium repo: https://github.com/zadam/trilium/issues/3181
Perhaps you could reach out to the OP there to see if they ever made any progress.

**Reply by elaiel** [reply]
I will do that, thanks :-)
